### DISCUSSION TITLE
Text-Based Scratch

### DISCUSSION THREAD

**Discussion by Zeboosta** [ScratchAddons/ScratchAddons#5157] [Ideas]
Make an addon that can make Scratch's code editor text-based in its entirety: This means NO palette, and the mouse wheel click toggling a block dropdown menu should toggle a keyword dropdown menu instead; new blocks will be made using a format "function exampleFunction (param) {
}"

Cloud variables will be made using the format "cloud var exampleVariable" and variables will be made using the format "var anotherVariable".

**Comment by BroJac5246** [comment]
> This means NO palette

That sort of reminds me of (the opposite of) code.org's solution, which is to keep the block palette and convert the block to text when the user lets go of it.
> mouse wheel click

Not everyone has a mouse wheel.

I doubt this would be implemented, but it's an interesting suggestion. I wonder if this could be combined with [leopard.js](https://leopardjs.com/) somehow.

**Reply by CST1229** [reply]
> Not everyone has a mouse wheel.

Devtools block injection (which I assume is what the user is referring to) is not required to use Scratch.

**Comment by Zeboosta** [comment]
Mouse click for those who actually use a mouse, it will do a keyword
dropdown,

**Comment by BroJac5246** [comment]
> Mouse click for those who actually use a mouse, it will do a keyword dropdown,

Why not just add a keyboard shortcut? Plenty of hotkeys would be free when using this addon, and that way anyone with a keyboard (so anyone using this addon) could benefit, not just those who use a mouse.
